src/prefect/tasks/filehandling/compression.py

```diff
@@ -0,0 +1,249 @@
+"""
+The tasks in this module can be used to handle file compression like zipping or
+unzipping files.
+"""
+import os
+import sys
+import zipfile
+from pathlib import Path
+from typing import Any, List, Union
+from zipfile import ZipFile
+
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.tasks.filehandling.base import FileBase
+
+
+class Unzip(FileBase):
+    """
+    Task for unzipping data.
+
+    Args:
+        - zip_file_path (Union[str, Path], optional): path to the zip file
+        - target_directory (Union[str, Path], optional): path to the target directory
+        - zip_password (str, optional): password for extraction. This must be utf-8 encoded.
+        - create_target_if_not_exists (bool, optional): Create the target directory if it
+            doesn't exist (default: False)
+        - use_filename_as_target_dir (bool, optional): Extracts data within a folder named
+            like the zip archive (without extension). If True, the basename of the zip archive will
+            be attached to the target_directory and may create a new folder. (default: False)
+        - remove_after_unzip (bool, optional): Removes the zip archive after extraction (default: False)
```

Comment:
There's a lot of options here, some of which seem unnecessary. Adding flexibility is good when useful, but can also confuse users and increases our maintenance burden. If I were writing this I'd go with:

```python
"""
- source (Union[str, Path], optional): the path to the zip file
- target_dir (Union[str, Path], optional): directory to extract the zip file into.
    If not provided, the current working directory will be used. This directory
    must already exist.
"""

The `run` method would support all of the above, as well as

```python
"""
- password (str, optional): password for unzipping a password-protected zip file.
"""
```

Supporting `password` in the run method alone better encourages users to store passwords in secrets rather than hardcoding the parameter in the constructor.

This design better matches the functionality of the `unzip` shell command (or `shutil.unpack_archive`), both existing interfaces a user is likely already familiar with. The removed options would better be handled by other tasks, or can be inferred automatically from the parameters above.